codalab/server/bundle_manager.py

```diff
@@ -120,7 +120,8 @@ def _run_iteration(self):
         self._fail_unresponsive_bundles()
 
     def _set_staged_status(self, bundle, staged_status):
-        self._model.update_bundle(bundle, {'metadata': {'staged_status': staged_status}})
+        if bundle.BUNDLE_TYPE == 'run':
```

Comment:
Shouldn't this function only be called when the bundle is a run?

Reply:
There is an edge case in which `make` bundles are staged: https://github.com/codalab/codalab-worksheets/blob/master/codalab/server/bundle_manager.py#L191-L196

I have removed this defensive check and updated the `MakeBundle` class to allow `staged_status` metadata.